src/ArcCommander/APIs/AssayAPI.fs

```diff
@@ -188,14 +188,53 @@ module AssayAPI =
 
         let assayFilepath = IsaModelConfiguration.tryGetAssayFilePath assayIdentifier arcConfiguration |> Option.get
 
-        let _, _, _, oldAssay = AssayFile.Assay.fromFile assayFilepath
+        // read assay metadata information from assay file
+        let _, _, _, oldAssayAssayFile = AssayFile.Assay.fromFile assayFilepath
+        
+        // read assay metadata information from investigation file
+        let noAssayFound, oldAssayInvestigationFile = 
+            let emptyAssay = Assays.fromString "" "" "" "" "" "" "" "" []
+            match investigation.Studies with
+            | Some studies -> 
+                match API.Study.tryGetByIdentifier studyIdentifier studies with
+                | Some study -> 
+                    match study.Assays with
+                    | Some assays -> 
+                        match API.Assay.tryGetByFileName assayFileName assays with
+                        | Some assay -> false, assay
+                        | None -> 
+                            if verbosity >= 1 then printfn "Assay with the identifier %s does not exist in the study with the identifier %s." assayIdentifier studyIdentifier
```

Comment:
I would call the comparison step, the query step and the edit step in here. So you don't have to open the investigation file twice and you don't have to model the state of the assay in the investigation file.